contracts/schemes/GenericSchemeMultiCall.sol

```diff
@@ -0,0 +1,167 @@
+            execute(_proposalId);
+        } else {
+            delete organizationProposals[_proposalId];
+            emit ProposalDeleted(address(avatar), _proposalId);
+        }
+
+        emit ProposalExecutedByVotingMachine(address(avatar), _proposalId, _decision);
+        return true;
+    }
+
+    /**
+    * @dev execution of proposals after it has been decided by the voting machine
+    * @param _proposalId the ID of the voting in the voting machine
+    */
+    function execute(bytes32 _proposalId) public {
+        MultiCallProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.exist, "must be a live proposal");
+        require(proposal.passed, "proposal must passed by voting machine");
+        proposal.exist = false;
+        bytes memory genericCallReturnValue;
+        bool success;
+        Controller controller = Controller(avatar.owner());
+
+        for (uint i = 0; i < proposal.contractToCall.length; i ++) {
+            (success, genericCallReturnValue) =
+            controller.genericCall(proposal.contractToCall[i], proposal.callData[i], avatar, proposal.value[i]);
+            require(success, "Proposal call failed");
+            emit ProposalCallExecuted(address(avatar), _proposalId, proposal.contractToCall[i], genericCallReturnValue);
+        }
+
+        delete organizationProposals[_proposalId];
+        emit ProposalDeleted(address(avatar), _proposalId);
+        emit ProposalExecuted(address(avatar), _proposalId);
+
+    }
+
+    /**
+    * @dev propose to call on behalf of the _avatar
+    *      The function trigger NewCallProposal event
+    * @param _contractToCall the contract to be called 
+    * @param _callData - The abi encode data for the call
+    * @param _value value(ETH) to transfer with the call
+    * @param _descriptionHash proposal description hash
+    * @return an id which represents the proposal
+    */
+
+    function proposeCall(address[] memory _contractToCall, bytes[] memory _callData, uint256[] memory _value, string memory _descriptionHash)
+    public
+    returns(bytes32)
+    {
+        require(
+            (_contractToCall.length == _callData.length) && (_contractToCall.length == _value.length),
+            "Wrong length of _contractToCall, _callData or _value arrays"
+        );
+        bytes32 proposalId = votingMachine.propose(2, voteParams, msg.sender, address(avatar));
+
+        organizationProposals[proposalId] = MultiCallProposal({
+            contractToCall: _contractToCall,
+            callData: _callData,
+            value: _value,
+            exist: true,
+            passed: false
+        });
+        proposalsInfo[address(votingMachine)][proposalId] = ProposalInfo({
+            blockNumber:block.number,
+            avatar:avatar
+        });
+
+        emit NewMultiCallProposal(address(avatar), proposalId, _callData, _value, _descriptionHash, _contractToCall);
+        return proposalId;
```

Comment:
can be removed

Reply:
Why removing it? We would still want to emit the event and return proposalId after creating the Proposal